annif/project.py

```diff
@@ -215,6 +225,21 @@ def suggest(self, text, backend_params=None):
         logger.debug("%d hits from backend", len(hits))
         return hits
 
+    def suggest_batch(self, corpus, backend_params=None):
+        """Suggest subjects for the given documents using batches of documents in their
+        operations when possible."""
+        if not self.is_trained:
+            if self.is_trained is None:
+                logger.warning("Could not get train state information.")
+            else:
+                raise NotInitializedException("Project is not trained.")
+        corpus = self.transform.transform_corpus(corpus)
+        logger.debug(
+            f"Suggesting subjects for a batch of {sum(1 for _ in corpus.documents)}"
```

Comment:
This can be quite inefficient, as it requires looping through a potentially very large corpus, which also has to be transformed. It would be better to avoid counting the documents, unless that information is essential in the debug message. Also, it could potentially change the behaviour of the suggest operation below, depending on how `corpus.documents` has been implemented.